src/HealthChecks.Network/DnsResolveHealthCheck.cs

```diff
@@ -32,12 +33,16 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
                 {
                     if (item.Resolutions == null || !item.Resolutions.Contains(ipAddress.ToString()))
                     {
-                        return new HealthCheckResult(context.Registration.FailureStatus, description: $"Ip Address {ipAddress} was not resolved from host {item.Host}");
+                        errorList.Add($"Ip Address {ipAddress} was not resolved from host {item.Host}");

```

Comment:
```suggestion
                        (errorList ??= new()).Add($"Ip Address {ipAddress} was not resolved from host {item.Host}");

```

Reply:
I would avoid memory allocations for green path.

Reply:
Do you mean, that errorList should only be initialized if there is an error?

Reply:
```
public static class EmptyList<T>
{
    internal static readonly List<T> Value = new List<T>();
}
```

And initialize it with

`List<string> errorList = EmptyList<string>.Value;`

With this helper class no additional memory is allocated.

Reply:
> Do you mean, that errorList should only be initialized if there is an error?

Yes.

> With this helper class no additional memory is allocated.

It does not work. Static instance will be used by all checks concurrently.